examples/demo-stf/src/sov-cli/main.rs

```diff
@@ -0,0 +1,222 @@
+use anyhow::Context;
+use borsh::BorshSerialize;
+use clap::Parser;
+use sov_modules_api::transaction::Transaction;
+use sov_modules_stf_template::RawTx;
+use std::fs;
+use std::fs::File;
+use std::io::Write;
+use std::path::{Path, PathBuf};
+
+use demo_stf::runtime::cmd_parser;
+
+use sov_modules_api::{
+    default_context::DefaultContext, default_signature::private_key::DefaultPrivateKey,
+    AddressBech32, PublicKey, Spec,
+};
+
+type C = DefaultContext;
+type Address = <C as Spec>::Address;
+
+/// Main entry point for CLI
+#[derive(Parser)]
+#[clap(version = "1.0", author = "Sovereign")]
+struct Cli {
+    #[clap(subcommand)]
+    /// Commands to perform operations
+    command: Commands,
+}
+
+/// Main commands
+#[derive(Parser)]
+enum Commands {
+    /// Serialize a call to a module.
+    /// This creates a dat file containing the serialized transaction
+    SerializeCall {
+        /// Path to the json file containing the private key of the sender
+        sender_priv_key_path: String,
+        /// Name of the module to generate the call.
+        /// Modules defined in your Runtime are supported.
+        /// (eg: Bank, Accounts)
+        module_name: String,
+        /// Path to the json file containing the parameters for a module call
```

Comment:
How do these json files get generated? Is there any way we can do that automatically as well?

Reply:
These json files would be generated manually based on the module the user wants to call. for example:
```
{
    "CreateToken": {
      "salt": 11,
      "token_name": "sov-test-token",
      "initial_balance": 1000,
      "minter_address": "...",
      "authorized_minters": ["..."]
    }
}
```
are you talking about perhaps letting the entire call be clapped so that a user can do something like:
-- module-call create-token 11 sov-test-token --initial-balance=1000 
or something similar?

Reply:
one good thing if we decide to go this route is that we wouldn't need json deser to be implemented for CallMessage. @bkolad ? could explore this in a different issue

Reply:
I think it is ok to do it in different PR